### PR TITLE
Fixes import abort when trying to import inital large repositories 

### PR DESCRIPTION
**Pull request by hcw70**
... since a duplicate change_id / bug_id is reported for table mantis_plugin_Source_bug_table.

I was trying to import a repository using gitlab plugin.

### CODE REVIEW

File: Source/Source.API.php [hunk 1]
```diff
@@ -997,7 +997,7 @@ function save_bugs( $p_user_id=null ) {
 		}
 
 		if ( count( $t_bugs_added ) > 0 ) {
-			$t_query = "INSERT INTO $t_bug_table ( change_id, bug_id ) VALUES ";
+			$t_query = "INSERT IGNORE INTO $t_bug_table ( change_id, bug_id ) VALUES ";
```
Comment (atrol) [inline Source/Source.API.php]: INSERT IGNORE is not supported by all databases that are supported by MantisBT.

**Comment by hcw70** [comment]
Ok, then this needs to be replaced by an "select ..." and "insert" then?

**Comment by dregad** [comment]
> a duplicate change_id / bug_id is reported

I'm a bit surprised, how could this happen ?

**Comment by hcw70** [comment]
@dregad Maybe because the message of the contains multiple matches with the bug pattern...?

**Comment by dregad** [comment]
> the message of the commit contains multiple matches with the bug pattern

Is that a statement of fact, or just a supposition ? 

Anyway, I find this unlikely because of array_unique() call: https://github.com/mantisbt-plugins/source-integration/blob/master/Source/Source.API.php#L969

**Comment by hcw70** [comment]
This was an assumption. When not using that fix, i get

    APPLICATION ERROR #401

    Database query failed. Error received from database was #1062: Duplicate entry '527' for key 'change_id' for the query: INSERT INTO mantis_plugin_Source_bug_table ( change_id, bug_id ) VALUES (?, ?), (?, ?).


    Please use the "Back" button in your web browser to return to the previous page. There you can correct whatever problems were identified in this error or select another action. You can also click an option from the menu bar to go directly to a new section.

when importing one of my repositories.

**Comment by hcw70** [comment]
Is there any detail i can supply to help fixing this bug?

**Comment by dregad** [comment]
Providing step-by-step instructions allowing someone to reproduce the problem would be a good starting point.

**Comment by dregad** [comment]
@hcw70 you have never provided any feedback, so I'll close this for now. Feel free to reopen it if you can provide additional information that would allow reproducing the problem.

**Pull request closed without merging**